src/core/opentelemetry.ml

```diff
@@ -1463,12 +1464,12 @@ module Metrics_callbacks = struct
       of metrics. It might be called regularly by the backend, in particular
       (but not only) when {!Collector.tick} is called. *)
   let register f : unit =
-    if !cbs_ = [] then
+    if AList.is_empty cbs_ then
       (* make sure we call [f] (and others) at each tick *)
         Collector.on_tick (fun () ->
-          let m = List.map (fun f -> f ()) !cbs_ |> List.flatten in
+          let m = List.map (fun f -> f ()) (AList.get cbs_) |> List.flatten in
```

Comment:
Is there actually another, non-benign race condition here that would work as follows?

1. 2 threads both call `register f`
2. They check `cbs_` at the "same" time.
3. So both call `Collector.on_tick`, resulting in two instances of this callback being registered, and thus erroneous repetitions of the registered metrics on each tick.

If so, then I think this would require a mutex to prevent?

Reply:
in general I'd expect registration/setup to be single threaded, as is often the case. But maybe a lock is safer if you're worried about it :)

Reply:
I mean, using `AList` is great, do keep it! I was specifically mentioning a separate `bool Atomic.t` in the context of initializing exactly once.

Reply:
Oh, I see! Makes sense. I'll make that change.